chapters/strings/finding-substrings.md

```diff
@@ -24,8 +24,14 @@ message.indexOf "This", 5
 message.lastIndexOf "This"
 # => 49
 
+# Count occurrences of a given string within a message
+String::occurrences = (search) ->
+  regex = new RegExp(search, 'g')
+  (@match(regex) or []).length
```

Comment:
This is not very performant. Your needlessly creating an empty array. How about this:

```coffee
regex.test(this)
```